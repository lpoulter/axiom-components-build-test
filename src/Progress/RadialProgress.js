import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './RadialProgress.css';

export default class RadialProgress extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string.isRequired,
    diameter: PropTypes.number.isRequired,
    size: PropTypes.string,
    sizeRem: PropTypes.string,
  };

  render() {
    const { children, className, diameter, size, sizeRem, ...rest } = this.props;
    const style = {};
    const classes = classnames(className, 'ax-radial-progress', {
      [`ax-radial-progress--${size}`]: size,
    });

    if (sizeRem) {
      style.width = sizeRem;
      style.height = sizeRem;
    }

    return (
      <Base { ...rest }
          Component="svg"
          className={ classes }
          style={ style }
          viewBox={ `0 0 ${ diameter } ${ diameter }` }>
        <g transform={ `translate(${ diameter / 2 }, ${ diameter / 2 }) rotate(-90)` }>
          { children }
        </g>
      </Base>
    );
  }
}
