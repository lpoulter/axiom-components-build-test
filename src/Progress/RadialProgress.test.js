import React from 'react';
import renderer from 'react-test-renderer';
import RadialProgress from './RadialProgress';

const getComponent = (props = {}) =>
  renderer.create(
    <RadialProgress className="lorem" diameter={ 75 } size="small" { ...props }>
      <svg />
    </RadialProgress>
  );

describe('RadialProgress', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with size', () => {
    ['small', 'medium', 'large'].forEach((size) => {
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
