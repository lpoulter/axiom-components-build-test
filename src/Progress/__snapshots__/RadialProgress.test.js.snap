// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadialProgress renders with defaultProps 1`] = `
<svg
  className="lorem ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 75 75"
>
  <g
    transform="translate(37.5, 37.5) rotate(-90)"
  >
    <svg />
  </g>
</svg>
`;

exports[`RadialProgress renders with size large 1`] = `
<svg
  className="lorem ax-radial-progress ax-radial-progress--large"
  style={Object {}}
  viewBox="0 0 75 75"
>
  <g
    transform="translate(37.5, 37.5) rotate(-90)"
  >
    <svg />
  </g>
</svg>
`;

exports[`RadialProgress renders with size medium 1`] = `
<svg
  className="lorem ax-radial-progress ax-radial-progress--medium"
  style={Object {}}
  viewBox="0 0 75 75"
>
  <g
    transform="translate(37.5, 37.5) rotate(-90)"
  >
    <svg />
  </g>
</svg>
`;

exports[`RadialProgress renders with size small 1`] = `
<svg
  className="lorem ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 75 75"
>
  <g
    transform="translate(37.5, 37.5) rotate(-90)"
  >
    <svg />
  </g>
</svg>
`;
